Feature/shuai add comment

shuai-deng opened the pull request:
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Commit:
add LogOffsetMetadata comment

Commit:
add log comment

Commit:
add log comment

Commit:
add log comment-二分查找

Commit:
lookup

Commit:
timeindex注释

Commit:
RequestChannel注释

Commit:
SocketServer 注释

Commit:
SocketServer 注释

Commit:
SocketServer 注释:KafkaRequestHandler代码流程

Commit:
请求处理全流程共6步：
KafkaRequestHandler：I/O 线程，负责处理 Processor 线程下发的 Request 对象。
KafkaRequestHandlerPool：创建和管理一组 KafkaRequestHandler 线程。
请求处理流程：总共分为 6 步。
1、Clients 或其他 Broker 通过 Selector 机制发起创建连接请求。
2、Processor 线程接收请求，并将其转换成可处理的 Request 对象。
3、Processor 线程将 Request 对象放入 Request 队列。
4、KafkaRequestHandler 线程从 Request 队列中取出待处理请求，并进行处理。
5、KafkaRequestHandler 线程将 Response 放回到对应 Processor 线程的 Response 队列。
6、Processor 线程发送 Response 给 Request 发送方。

Commit:
KafkaApis 添加注释

Commit:
KafkaApis 添加注释

Commit:
ControllerContext 添加注释

Commit:
ControllerContext 添加注释

Commit:
ControllerChannelManager 添加注释

Commit:
ControllerChannelManager 添加注释

Commit:
ControllerEventManager 添加注释

Commit:
ControllerEventManager 添加注释

Commit:
ControllerEventManager 添加注释

Commit:
Controller选举过程

divijvaidya commented:
Duplicate of https://github.com/apache/kafka/pull/13730 

@DengShuaiSimon As I mentioned in the previous PR, unfortunately, we cannot merge this change. As per ASF code of conduct policy [1], we expect all development of Apache projects (including documentation) to take place in English language. You are welcome to submit these changes again in English language. Please feel free to comment if you have questions.

[1] https://www.apache.org/foundation/policies/conduct.html#diversity-statement

Commit:
Controller集群成员管理

Commit:
Controller成员信息管理

Commit:
注释：TopicDeletionManager： Topic是怎么被删除的？

Commit:
注释：副本状态机实现原理【ReplicaStateMachine】： 1：转换到 NewReplica 状态

Commit:
注释：副本状态机实现原理【ReplicaStateMachine】

Commit:
注释：PartitionStateMachine：分区状态转换实现

Commit:
注释：PartitionStateMachine：Kafka 分区状态机的构造原理和工作机制。
PartitionStateMachine 是 Kafka Controller 端定义的分区状态机，负责定义、维护和管理合法的分区状态转换。
每个 Broker 启动时都会实例化一个分区状态机对象，但只有 Controller 所在的 Broker 才会启动它。
Kafka 分区有 4 类状态，分别是 NewPartition、OnlinePartition、OfflinePartition 和 NonExistentPartition。其中 OnlinPartition 是分区正常工作时的状态。NewPartition 是未初始化状态，处于该状态下的分区尚不具备选举 Leader 的资格。
Leader 选举有 4 类场景，分别是 Offline、Reassign、Preferrer Leader Election 和 ControlledShutdown。每类场景都对应于一种特定的 Leader 选举策略。
handleStateChanges 方法是主要的入口方法，下面调用 doHandleStateChanges 私有方法实现实际的 Leader 选举功能。

divijvaidya commented:
I am closing this PR. Please refer to the comment above for the reason - https://github.com/apache/kafka/pull/13778#issuecomment-1567994161
